0.12 force-unlock, segmentation violation

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Terraform Version


```
Terraform v0.12.0-beta1
+ provider.aws v1.60.0-dev20190216h00-dev
+ provider.random v3.0.0-dev20190216h01-dev
```

### Crash Output
```
> terraform-12 force-unlock 4ce74129-cc24-209d-438e-0d340c55381e -force
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x21d0603]

goroutine 1 [running]:
github.com/hashicorp/terraform/command.(*Meta).loadBackendConfig(0xc0004be180, 0x7ffeefbffa76, 0x6, 0x7ffeefbffa76, 0x6, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_config.go:175 +0x53
github.com/hashicorp/terraform/command.(*UnlockCommand).Run(0xc0004be180, 0xc0000900a0, 0x2, 0x2, 0xc0004547f0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/unlock.go:53 +0x254
github.com/mitchellh/cli.(*CLI).Run(0xc0004a4500, 0xc0004a4500, 0xc00041dd20, 0x1)
	/opt/gopath/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x207
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:36 +0x2e



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Expected Behavior
LOCK_ID removed

### Actual Behavior
signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x21d0603

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`
3. interrupt terraform execution to leave state locked
4. `terraform-12 force-unlock LOCK_ID -force`

--- a/command/meta_config.go
+++ b/command/meta_config.go
@@ -173,6 +173,9 @@ func (m *Meta) dirIsConfigPath(dir string) bool {
 // directory even if loadBackendConfig succeeded.)
 func (m *Meta) loadBackendConfig(rootDir string) (*configs.Backend, tfdiags.Diagnostics) {
 	mod, diags := m.loadSingleModule(rootDir)
+	if diags.HasErrors() {
+		return nil, diags
+	}
 	return mod.Backend, diags
 }
 

--- a/command/unlock_test.go
+++ b/command/unlock_test.go
@@ -49,6 +49,16 @@ func TestUnlock(t *testing.T) {
 	if code := c.Run(args); code != 1 {
 		t.Fatalf("bad: %d\n%s\n%s", code, ui.OutputWriter.String(), ui.ErrorWriter.String())
 	}
+
+	// make sure we don't crash with arguments in the wrong order
+	args = []string{
+		"LOCK_ID",
+		"-force",
+	}
+
+	if code := c.Run(args); code != 1 {
+		t.Fatalf("bad: %d\n%s\n%s", code, ui.OutputWriter.String(), ui.ErrorWriter.String())
+	}
 }
 
 // Newly configured backend

